hummingbot/connector/exchange/polkadex/polkadex_order_book_data_source.py

```diff
@@ -0,0 +1,129 @@
+
+    async def _parse_order_book_diff_message(self, raw_message: Dict[str, Dict[str, List[Dict[str, str]]]],
+                                             message_queue: asyncio.Queue):
+        """
+               {
+                   'websocket_streams': {
+                       'data': '[{"side":"Ask","price":5554500000000,"qty":7999200000000 ,"seq":20}]'
+                   }
+               }
+        """
+        diff_message = PolkadexOrderbook.diff_message_from_exchange(raw_message)
+        message_queue.put_nowait(diff_message)
+
+    async def _order_book_snapshot(self, trading_pair: str) -> OrderBookMessage:
+        result: List[Dict[str, Any]] = await get_orderbook(trading_pair, None, None, self._connector.host,
+                                                           self._connector.user_proxy_address)
+        snapshot_timestamp: float = time.time()
+        snapshot_msg: OrderBookMessage = PolkadexOrderbook.snapshot_message_from_exchange(
+            result,
+            snapshot_timestamp,
+            metadata={"trading_pair": trading_pair}
+        )
+        return snapshot_msg
+
+    def on_recent_trade_callback(self, message, trading_pair):
+        # Expected structure
+        # {'type': 'TradeFormat', 'm': 'PDEX-3', 'p': '2', 'vq': '20', 'q': '10', 'tid': '111', 't': 1664193952989, 'sid': '16'}
+        new_message = {}
+        new_message["data"] = []
+        message = message["websocket_streams"]["data"]
+        message = json.loads(message)
+        change = {}
+        change["p"] = p_utils.parse_price_or_qty(message['p'])
+        change["q"] = p_utils.parse_price_or_qty(message["q"])
+        change["vq"] = p_utils.parse_price_or_qty(message["vq"])
+        change["t"] = p_utils.parse_price_or_qty(message["t"])
+        change["tid"] = p_utils.parse_price_or_qty(message["tid"])
+        change["m"] = trading_pair
+        new_message["data"].append(change)
+        self._message_queue[self._trade_messages_queue_key].put_nowait(new_message)
+
+    def on_ob_increment(self, message, trading_pair):
+        # {
+        #       "websocket_streams": {
+        #         "data":
+        #         '{"type":"IncOB","changes":[["Ask","3","2",123]]}'
+        #       }
+        #     }
+        message = message["websocket_streams"]["data"]
+        message = json.loads(message)
+        message = message["changes"]
+        new_message = {}
+        new_message["side"] = message[0][0]
+        new_message["price"] = message[0][1]
+        new_message["qty"] = message[0][2]
+        new_message["id"] = message[0][3]
+        new_message["market"] = trading_pair
+        self._message_queue[self._diff_messages_queue_key].put_nowait(new_message)
+
+    async def listen_for_subscriptions(self):
+        transport = AppSyncWebsocketsTransport(url=self._connector.wss_url, auth=self._connector.auth)
+        tasks = []
+        async with Client(transport=transport, fetch_schema_from_transport=False) as session:
```

Comment:
how are you handling cancel error, timeout or exceptions?

Reply:
Wrapped it in `try/except`